sdk/python/feast/infra/offline_stores/bigquery.py

```diff
@@ -310,6 +325,25 @@ def _execute_query(
         block_until_done(client=self.client, bq_job=bq_job, timeout=timeout)
         return bq_job
 
+    def persist(self, storage: SavedDatasetStorage) -> "BigQueryRetrievalJob":
+        assert isinstance(storage, SavedDatasetBigQueryStorage)
```

Comment:
Is the BigQuery offline store can only persist the SavedDataset into BigQuery, why making it possible to pass any `SavedDatasetStorage` and not infer it ?

Reply:
As described above, that's merely for type checking: good for verification on development and could be helpful when using Feast in some kind of IDE.

Reply:
If I understand correctly, you suggesting just plain dict:
```
fs.get_historical_features(..., save_as={'storage': {'table_ref': '...'}})
```
vs proposed API
```
fs.get_historical_features(..., save_as=SavedDatasetOpts(storage=SavedDatasetBigQueryStorage(table_ref=...))
```
just to free you from remembering those dict keys.

Reply:
> If I understand correctly, you suggesting just plain dict

Or sending parameters right away. Something like
```python
get_historical_features(..., save_as=SavedDataset(table_ref="...")) # Example 1
get_historical_features(..., save_as=SavedDataset(table_ref="...", table_ttl=timedelta(days=30))) # Example 2
get_historical_features(..., save_as=SavedDataset(file_path="...")) # Example 3

# Under the hood
@dataclass
class SavedDataset:
    table_ref: Optional[str] = None
    table_ttl: Optional[timedelta] = None
    file_path: Optional[str] = None

def get_historical_features(..., save_as: SavedDatasetOpts):
    if isinstance(offline_store, BigQuery):
        return SavedDatasetBigQueryStorage(table_ref=save_as.table_ref, table_ttl=save_as.table_ttl)
```

Unsure if that makes it more confusing **or** easier to use for our users TBH 🤔

Reply:
I think subclasses is a better approach:
1. Easier to extend (add new kind of storage)
2. You can have mandatory & optional parameters, whereas with a single class they all are optional.

Plus I believe this approach (more strict types) is used in various places in Feast, so keeping things consistent is good.

Reply:
> I think subclasses is a better approach

Sounds good to me 👍